src/transactions/transactionBytes.js

```diff
@@ -12,323 +12,252 @@
-		for (let i = 0; i < keysgroupBuffer.length; i++) {
-			bb.writeByte(keysgroupBuffer[i]);
-		}
-		bb.flip();
+export function getAssetDataForDelegateTransaction({ delegate }) {
+	checkRequiredFields(['username'], delegate);
+	const { username } = delegate;
+	return Buffer.from(username, 'utf8');
+}
 
-		bb.toBuffer();
-		const multiSigBuffer = new Uint8Array(bb.toArrayBuffer());
+/**
+ * @method getAssetDataForVotesTransaction
+ * @param {Object} transactionAsset
+ * @return {Buffer}
+ */
 
-		return {
-			assetBytes: multiSigBuffer,
-			assetSize: multiSigBuffer.length,
-		};
+export function getAssetDataForVotesTransaction({ votes }) {
+	if (!Array.isArray(votes)) {
+		throw new Error('votes parameter must be an Array.');
 	}
+	return Buffer.from(votes.join(''), 'utf8');
+}
 
-	/**
-	 * @method isDappTransaction
-	 * @return {object}
-	 */
-
-	function isDappTransaction() {
-		const dapp = transaction.asset.dapp;
-		let buf = Buffer.from(dapp.name);
-
-		if (dapp.description) {
-			const descriptionBuf = Buffer.from(dapp.description);
-			buf = Buffer.concat([buf, descriptionBuf]);
-		}
-
-		if (dapp.tags) {
-			const tagsBuf = Buffer.from(dapp.tags);
-			buf = Buffer.concat([buf, tagsBuf]);
-		}
-
-		if (dapp.link) {
-			buf = Buffer.concat([buf, Buffer.from(dapp.link)]);
-		}
+/**
+ * @method getAssetDataForMultisignatureTransaction
+ * @param {Object} transactionAsset
+ * @return {Buffer}
+ */
 
-		if (dapp.icon) {
-			buf = Buffer.concat([buf, Buffer.from(dapp.icon)]);
-		}
+export function getAssetDataForMultisignatureTransaction({ multisignature }) {
+	checkRequiredFields(['min', 'lifetime', 'keysgroup'], multisignature);
+	const { min, lifetime, keysgroup } = multisignature;
+	const minBuffer = Buffer.alloc(1, min);
+	const lifetimeBuffer = Buffer.alloc(1, lifetime);
+	const keysgroupBuffer = Buffer.from(keysgroup.join(''), 'utf8');
 
-		const bb = new ByteBuffer(4 + 4, true);
-		bb.writeInt(dapp.type);
-		bb.writeInt(dapp.category);
-		bb.flip();
+	return Buffer.concat([minBuffer, lifetimeBuffer, keysgroupBuffer]);
+}
 
-		buf = Buffer.concat([buf, bb.toBuffer()]);
+/**
+ * @method getAssetDataForDappTransaction
+ * @param {Object} transactionAsset
+ * @return {Buffer}
+ */
 
-		return {
-			assetBytes: buf,
-			assetSize: buf.length,
-		};
-	}
+export function getAssetDataForDappTransaction({ dapp }) {
+	checkRequiredFields(['name', 'link', 'type', 'category'], dapp);
+	const { name, description, tags, link, icon, type, category } = dapp;
+	const dappNameBuffer = Buffer.from(name, 'utf8');
+	const dappLinkBuffer = Buffer.from(link, 'utf8');
+	const dappTypeBuffer = Buffer.alloc(4, type);
+	const dappCategoryBuffer = Buffer.alloc(4, category);
+
+	const dappDescriptionBuffer = description ? Buffer.from(description, 'utf8') : Buffer.alloc(0);
+	const dappTagsBuffer = tags ? Buffer.from(tags, 'utf8') : Buffer.alloc(0);
+	const dappIconBuffer = icon ? Buffer.from(icon, 'utf8') : Buffer.alloc(0);
```

Comment:
Since we're in the context of a dapp transaction, better to remove the `dapp` prefix from all of these for readability (as in multisignature function above).